editor.planx.uk/src/@planx/components/FindProperty/Public/index.tsx

```diff
@@ -42,27 +45,48 @@ export const FETCH_BLPU_CODES = gql`
 
 type Props = PublicProps<FindProperty>;
 
+interface PickOSAddressProps {
+  title?: string;
+  description?: string;
+  allowNewAddresses?: boolean;
+  initialPostcode?: string;
+  initialSelectedAddress?: SiteAddress;
+  teamSettings?: TeamSettings;
+  id?: string;
+  setAddress: React.Dispatch<React.SetStateAction<SiteAddress | undefined>>;
+  setPage: React.Dispatch<React.SetStateAction<"os-address" | "new-address">>;
+}
+
 export default Component;
 
 function Component(props: Props) {
   const previouslySubmittedData = props.previouslySubmittedData?.data;
+  const previouslySubmittedAddressSource =
+    props.previouslySubmittedData?.data?._address?.source;
+  const startPage =
+    previouslySubmittedAddressSource === "proposed"
+      ? "new-address"
+      : "os-address";
+
+  const [page, setPage] = useState<"os-address" | "new-address">(startPage);
   const [address, setAddress] = useState<SiteAddress | undefined>();
   const [localAuthorityDistricts, setLocalAuthorityDistricts] = useState<
     string[] | undefined
   >();
   const [regions, setRegions] = useState<string[] | undefined>();
+
   const flow = useStore((state) => state.flow);
   const team = fetchCurrentTeam();
 
-  // if we have an address point, check which local authority district(s) & region it's located in via Digital Land
+  // Use the address point to fetch the Local Authority District(s) & region via Digital Land
   let options = new URLSearchParams({
-    entries: "all", // includes historic
+    entries: "all", // includes historic for pre-merger LADs (eg Wycombe etc for Uniform connector mappings)
     geometry: `POINT(${address?.longitude} ${address?.latitude})`,
     geometry_relation: "intersects",
     limit: "100",
   });
   options.append("dataset", "local-authority-district");
-  options.append("dataset", "region");
+  options.append("dataset", "region"); // proxy for Greater London Authority (GLA) boundary
```

Comment:
I did check if Digital Land has a postcode boundary file we could additionally query here in cases where the address source is "proposed", but alas they do not. But I don't think this should be a blocker, just would have been nice to have.